pkg/chain/ethereum/beacon.go

```diff
@@ -169,3 +185,7 @@ func (bc *BeaconChain) UpdateOperatorStatus() error {
 	_, err := bc.randomBeacon.UpdateOperatorStatus(bc.key.Address)
 	return err
 }
+
+func (bc *BeaconChain) StakeMonitor() (chain.StakeMonitor, error) {
+	return newStakeMonitor(bc, bc.Chain), nil
+}
```

Comment:
I am not convinced that per-application `StakeMonitor` is the right choice. `StakeMonitor` has just one function, instead of having `StakeMonitor()` for Beacon and TBTC chain, we can have a per-application `IsOperatorRecognized` function, and the `firewall.go` would expect two chains: Beacon and TBTC.

Alternatively, we can modify `Connect` function in `ethereum.go` to return `StakeMonitor` instance and the implementation, under the hood, would check operator<>staking provider mapping for `RandomBeacon`, `WalletRegistry`, and then called function from `threshold.go` (only one call compared to two calls to this function in the previous option) to see if the staking provider has roles assigned.

Reply:
Ok, thanks for the clarification! That makes perfect sense!

Reply:
Done in 1c58d940acf1a0369515201f4d63a6039de3f2f9.